### PR TITLE
Refactor Points and Rows to dedicated packages

### LINKED ISSUE #3447: Go client has too many dependencies
In upgrading `"github.com/influxdb/influxdb/client"` from 32846 (v0.8.0-rc.3-9-g328466) to 8b3219 (v0.9.1) I have to add dependencies:

```
golang.org/x/crypto/bcrypt
golang.org/x/crypto/blowfish
github.com/influxdb/influxdb/tsdb
github.com/influxdb/influxdb/toml
github.com/influxdb/influxdb/snapshot
github.com/influxdb/influxdb/meta
github.com/influxdb/influxdb/influxql
github.com/hashicorp/raft-boltdb
github.com/hashicorp/raft
github.com/hashicorp/go-msgpack/codec
github.com/golang/glog
github.com/gogo/protobuf/proto
github.com/boltdb/bolt
github.com/armon/go-metrics
```

These are not dependencies of the v0.8.0-rc.3-9-g328466 client. Why do I have to depend on a database, a raft implementation, a configuration language (among other things) to use the client? 

`"github.com/influxdb/influxdb/influxql"` and`"github.com/influxdb/influxdb/tsdb"` are the only two packages directly imported into the client. Is it possible to break up those packages so client doesn't have such a huge dependency graph?

### PR DESCRIPTION
**Pull request by corylanou**
Fixes #3447

### CODE REVIEW

File: client/influxdb.go [hunk 1]
```diff
@@ -13,8 +13,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/influxdb/influxdb/influxql"
-	"github.com/influxdb/influxdb/tsdb"
+	"github.com/influxdb/influxdb/models"
```
Comment (otoolep) [inline client/influxdb.go]: `models`?! We're building an ORM! :-)

**Comment by otoolep** [comment]
+1, don't see any issues doing something like this.

File: client/influxdb.go (hunk 1 above)
Comment (corylanou) [inline client/influxdb.go]: lol, it's the only name I could come up with.  I'm sure @benbjohnson will still hate it :smiling_imp:

**Comment by corylanou** [comment]
@jwilder Can you take a look over this when you get a chance.  I'm not super excited that I pulled the escaping to it's own package, but it was the only way to really decouple everything from `tsdb`.  Also, I'm not really fond of the `escape` package name.  `escape.Unescape` sounds like I'm doing a u-turn.

**Comment by jwilder** [comment]
:+1:

**Pull request merged**